manager/ui/war/plugins/api-manager/ts/api/api.ts

```diff
@@ -196,4 +223,53 @@ module Apiman {
                 });
             };
         }]);
+
+
+    export var ApiDeleteModalCtrl = _module.controller('ApiDeleteModalCtrl', function ($location,
+                                                                                       $rootScope,
+                                                                                       $scope,
+                                                                                       $uibModalInstance,
+                                                                                       ActionSvcs,
+                                                                                       Configuration,
+                                                                                       PageLifecycle,
+                                                                                       api,
+                                                                                       params) {
+
+        $scope.api = api;
+        $scope.params = params;
+
+        $scope.confirmApiName = '';
+
+        // Used for enabling/disabling the submit button
+        $scope.okayToDelete = false;
+
+        $scope.typed = function () {
+            // For user convenience, compare lower case values so that check is not case-sensitive
+            $scope.okayToDelete = ($scope.confirmApiName.toLowerCase() === api.name.toLowerCase());
+        };
+
+        // Yes, delete the API
+        $scope.yes = function () {
+            console.log('Preparing for delete.');

```

Comment:
+1 will update this in the morning. I'm off but will be around and this should be a quick change.